plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php

```diff
@@ -72,85 +71,94 @@ public function articleView($hookName, $args)
             return;
         }
 
+        $publication = $article->getCurrentPublication();
+        $publicationLocale = $publication->getData('locale');
+        $articleBestId = $publication->getData('urlPath') ?? $article->getId();
+
         $templateMgr = TemplateManager::getManager($request);
         $templateMgr->addHeader('dublinCoreSchema', '<link rel="schema.DC" href="http://purl.org/dc/elements/1.1/" />');
 
-        $i = 0;
-        if ($sponsors = $article->getSponsor(null)) {
-            foreach ($sponsors as $locale => $sponsor) {
-                $templateMgr->addHeader('dublinCoreSponsor' . $i++, '<meta name="DC.Contributor.Sponsor" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($sponsor)) . '"/>');
+        if ($supportingAgencies = $publication->getData('supportingAgencies')) {
+            foreach ($supportingAgencies as $locale => $localeSupportingAgencies) {
+                foreach ($localeSupportingAgencies as $i => $supportingAgency) {
+                    $templateMgr->addHeader('dublinCoreSponsor' . $locale . $i++, '<meta name="DC.Contributor.Sponsor" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars($supportingAgency) . '"/>');
+                }
             }
         }
 
-        $i = 0;
-        if ($coverages = $article->getCoverage(null)) {
+        if ($coverages = $publication->getData('coverage')) {
             foreach ($coverages as $locale => $coverage) {
-                $templateMgr->addHeader('dublinCoreCoverage' . $i++, '<meta name="DC.Coverage" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($coverage)) . '"/>');
+                if (!empty($coverage)) {
+                    $templateMgr->addHeader('dublinCoreCoverage' . $locale, '<meta name="DC.Coverage" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($coverage)) . '"/>');
+                }
             }
         }
 
-        $i = 0;
-        foreach ($publication->getData('authors') as $author) {
-            $templateMgr->addHeader('dublinCoreAuthor' . $i++, '<meta name="DC.Creator.PersonalName" content="' . htmlspecialchars($author->getFullName(false)) . '"/>');
+        if ($authors = $publication->getData('authors')) {
```

Comment:
Is this extra `if` necessary? `$publication->getData('authors')` should always be a collection; if it's empty, that's fine, no authors get added to the DC metadata.

Reply:
The metadata that can be enabled and disabled, like subjects, keywords and type, can be null, so for them I left this `if`.